src/vc/explain.rs

```diff
@@ -205,6 +218,23 @@ fn explain_park_induction(
     Ok(invariant.clone())
 }
 
+fn explain_unroll(vcgen: &mut Vcgen, body: &Block, args: &[Expr]) -> Result<Expr, Diagnostic> {
```

Comment:
I think there's a mistake here. The unrolling proof rule does not just run the body `k` times, but rather runs the `if cond { body; nesting } else {}` statement `k` times, nested inside `nesting`.

You can either do this manually here, or maybe just reuse the transformation code from the `unroll` mod and compute its vc here.